pkg/provider/azure_vmss_cache.go

```diff
@@ -286,11 +291,22 @@ func (ss *scaleSet) newAvailabilitySetNodesCache() (*azcache.TimedCache, error)
 
 			for _, vm := range vmList {
 				if vm.Name != nil {
-					localCache.Insert(*vm.Name)
+					vmNames.Insert(*vm.Name)
 				}
 			}
 		}
 
+		// store all the node names in the cluster when the cache data was created.
+		nodeNames, err := ss.GetNodeNames()
+		if err != nil {
+			return nil, err
+		}
+
+		localCache := availabilitySetEntry{
+			vmNames:   vmNames,
+			nodeNames: nodeNames,
```

Comment:
this doesn't look right. The nodeNames here include all nodes, including VMAS and VMSS

Reply:
It actually includes all nodes intentionally. The idea is that we only want to fetch the VM list if we absolutely have to in order to know if it's vmas or vmss. If the node is new to the cluster, there is a chance it wasn't added in the cache last time we fetched the VM list, so we have to fetch the list again. If it was already in the cluster last time we checked the cache, we don't need to check again, because if it was VM it would be in the cache. If it's a cache miss we don't need to check again, it's safe to assume it's VMSS.